File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    count = 0
    if s[0] == '(' and s[-1] == ')': 
        for char in s:
            if char  == '(':
                count += 1
            else:
                count -= 1
            if count < 0:
                return False
        if count == 0:
            return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int count = 0;
        boolean answer = false;
        if(s.charAt(0) == '(' && s.charAt(s.length()-1) == ')'){
            for (int i = 0; i < s.length(); i++){
                if(s.charAt(i) == '('){
                    count++;
                }else if (s.charAt(i) == ')'){
                    count--;
                }if(count < 0){
                    return false;
                }
            }
            if(count == 0){
                return true;
            }
        }
        return answer;
    }
}
